src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/ProjectDebuggerProvider.cs

```diff
@@ -25,29 +25,33 @@ namespace Microsoft.VisualStudio.ProjectSystem.VS.Debug
     [AppliesTo(ProjectCapability.LaunchProfiles)]
     internal class ProjectDebuggerProvider : DebugLaunchProviderBase, IDeployedProjectItemMappingProvider
     {
-
         /// <summary>
         /// Constructors. Unit test one is 2nd
         /// </summary>
         [ImportingConstructor]
-        public ProjectDebuggerProvider(ConfiguredProject configuredProject, ILaunchSettingsProvider launchSettingsProvider)
+        public ProjectDebuggerProvider(ConfiguredProject configuredProject, ILaunchSettingsProvider launchSettingsProvider, VsService<SVsShellDebugger, IVsDebugger4> vsDebuggerService)
             : base(configuredProject)
         {
             LaunchSettingsProvider = launchSettingsProvider;
+            _vsDebuggerService = vsDebuggerService;
 
             // We want it sorted so that higher numbers come first (is the default for these collections but explicitly expressed here)
             ProfileLaunchTargetsProviders = new OrderPrecedenceImportCollection<IDebugProfileLaunchTargetsProvider>(ImportOrderPrecedenceComparer.PreferenceOrder.PreferredComesFirst,
                                                                                                                     configuredProject.UnconfiguredProject);
         }
 
         public ProjectDebuggerProvider(ConfiguredProject configuredProject, ILaunchSettingsProvider launchSettingsProvider,
-                                       OrderPrecedenceImportCollection<IDebugProfileLaunchTargetsProvider> providers)
+                                       OrderPrecedenceImportCollection<IDebugProfileLaunchTargetsProvider> providers,
+                                       VsService<SVsShellDebugger, IVsDebugger4> vsDebuggerService)
```

Comment:
Gonna stop commenting now, but should be using the interface in most/all situations.